docs/examples/Reinforcement_Learning_Sensor_Manager.py

```diff
@@ -0,0 +1,1002 @@
+
+from stonesoup.dataassociator.neighbour import GNNWith2DAssignment
+data_associator = GNNWith2DAssignment(hypothesiser)
+
+from stonesoup.sensormanager.reward import UncertaintyRewardFunction
+reward_function = UncertaintyRewardFunction(predictor=predictor, updater=updater)
+
+# %%
+# Reinforcement Learning
+# ----------------------
+# Reinforcement learning is when an intelligent agents make decisions to maximise a cumulative reward. First, the agent
+# must train in an environment to create a policy, which determines which actions to take. During training, the agent
+# makes decisions and receives rewards, which is uses to optimise the policy.
+# 
+# .. image:: ../_static/rl_training.png
+#   :width: 800
+#   :alt: Illustration of sequential actions and measurements
+# 
+# Once training has completed, the policy can be exploited to gain rewards.
+# 
+
+# %%
+# Design Environment
+# ------------------
+# An environment is needed for the RL agent to learn in. There are resources online for how to design these [#]_.
+# 
+# For this tutorial, a pre-designed environment has been created for you to go through.
+# In this example, the action space is equal to the number of targets in the simulation, so at each time step, the
+# sensor can select one target to look at.
+# The :class:`~.UncertaintyRewardFunction` is used to calculate the reward obtained for each step in the environment.
+# The trace of the covariances for each object is used as the observation for the agent to learn from - it should learn
+# to select targets with a larger covariance (higher uncertainty).
+
+from abc import ABC
+import numpy as np
+import copy
+from ordered_set import OrderedSet
+
+from stonesoup.sensor.action.dwell_action import DwellActionsGenerator
+from stonesoup.functions import mod_bearing
+
+from tf_agents.environments import py_environment
+from tf_agents.specs import array_spec
+from tf_agents.trajectories import time_step as ts
+from tf_agents.environments import utils
+
+
+class StoneSoupEnv(py_environment.PyEnvironment, ABC):
+    """Example reinforcement learning environment. Environments must contain __init__, _reset,
+    _step, and generate_action methods
+    """
+
+    def __init__(self):
+        super().__init__()
+        # Action size is number of targets
+        self._action_spec = array_spec.BoundedArraySpec(
+            shape=(), dtype=np.int32, minimum=0, maximum=ntruths-1, name='action')
+        # Observation size is also number of targets
+        self.obs_size = ntruths
+        self._observation_spec = array_spec.BoundedArraySpec(
+            shape=(self.obs_size,), dtype=np.float32, name='observation')
+        self._episode_ended = False
+        self.max_episode_length = time_max
+        self.current_step = 0
+        self.start_time = start_time
+        # Use deepcopy to prevent the original sensor/tracks being changed
+        self.sensor = copy.deepcopy(sensorA)
```

Comment:
Don't get this. Don't you need to pass `sensorA` and `TracksA`?

Reply:
`sensorA` and `TracksA` are passed to the sensor manager, but in the environment, we make copies of them to train on. I added a note about this in the text.